apps/internal/base/base.go

```diff
@@ -83,14 +83,28 @@ type AcquireTokenOnBehalfOfParameters struct {
 // AuthResult contains the results of one token acquisition operation in PublicClientApplication
 // or ConfidentialClientApplication. For details see https://aka.ms/msal-net-authenticationresult
 type AuthResult struct {
-	Account        shared.Account
-	IDToken        accesstokens.IDToken
-	AccessToken    string
-	ExpiresOn      time.Time
-	GrantedScopes  []string
-	DeclinedScopes []string
+	Account            shared.Account
+	IDToken            accesstokens.IDToken
+	AccessToken        string
+	ExpiresOn          time.Time
+	GrantedScopes      []string
+	DeclinedScopes     []string
+	AuthResultMetadata AuthResultMetadata
 }
 
+// AuthResultMetadata which contains meta data for the AuthResult
+type AuthResultMetadata struct {
+	TokenSource TokenSource
+}
+
+type TokenSource int
+
+// These are all the types of token flows.
+const (
+	IdentityProvider TokenSource = 0
```

Comment:
The constant with value 0 becomes the default because underlying type is `int`, whose zero value is 0:
```go
ar := AuthResult{}
ar.Metadata.TokenSource == IdentityProvider // true
```
This may be irrelevant because the field is always set explicitly, however if you want a different default and/or a "source unknown" constant, that should have value `0`.

Reply:
Updated with the suggestion